src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -571,6 +609,7 @@ public void send(String method, Object... args) {
 
         sendHubMessage(streamInvocationMessage);
         Observable<T> observable = subject.doOnSubscribe((subscriber) -> subscriptionCount.incrementAndGet());
+        launchStreams(streamIds);
```

Comment:
Can the new logic that calls checkUploadStream/launchStreams be consolidated into the sendHubMessage method? Perhaps one that specialized in InvocationMessages.

Reply:
Updated with a `sendInvocationMessage` method for `send` `invoke` and `stream` that handles the logic for upload streams as well.